drivers/slipdev/slipdev.c

```diff
@@ -155,30 +118,59 @@ static int _send(netdev_t *netdev, const struct iovec *vector, unsigned count)
 static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
 {
     slipdev_t *dev = (slipdev_t *)netdev;
-    int res, idx = cib_peek(&dev->pktfifo_idx);
+    int res = 0;
 
     (void)info;
-    if (idx < 0) {
-        return -EFAULT;
-    }
     if (buf == NULL) {
         if (len > 0) {
-            /* drop packet */
-            cib_get(&dev->pktfifo_idx);
-            /* and remove data */
-            res = ringbuffer_remove(&dev->inbuf, len);
-        }
-        else {
-            res = dev->pktfifo[idx];
+            /* remove data */
+            for (; len > 0; len--) {
+                tsrb_get_one(&dev->inbuf);
+            }
+        } else {
+            /* the user was warned not to use a buffer size > `INT_MAX` ;-) */
+            res = (int)tsrb_avail(&dev->inbuf);
         }
     }
-    else if (len < dev->pktfifo[idx]) {
+    else if (len < tsrb_avail(&dev->inbuf)) {
```

Comment:
Is theoretically possible to have multiple packets in the tsrb at when this is checked? If a byte arrives in the tsrb between the `_recv()` call to get the expected buffer size and the `_recv()` call to copy the buffer to the pktsnip, this would evaluate to true and return an error.

Reply:
Additional problem clarification: If the above is possible, gnrc_netif_raw would return, but the current packet in the tsrb would never get removed, and would instead be returned as a received packet when the next packet is added to the tsrb (, as that one would trigger the next `SLIP_END`; `NETDEV_EVENT_RX_COMPLETE` etc sequence).

Reply:
Yes, I have fixed that now. See my latest commit.